modules/safer_mysql/README.md

```diff
@@ -0,0 +1,250 @@
+level of access correspond to the `roles/cloudsql.admin` and
+`roles/cloudsql.client`.
+
+`roles/cloudsql.admin`:
+
+-   Modifications to the instance configurations should be very rare, we suggest
+    to assign such permissions only to automation users (e.g.,
+    Terraform run as part of a CI/CD pipeline), or as part of breakglass access
+    to the project's resources.
+
+`roles/cloudsql.client`
+
+-   Service accounts of applications can be bound directly to the role (e.g.,
+    `xxxx-prod@xxxx.iam.gserviceaccont.com`)
+
+-   Humans that need to access the data for debugging purposes should get the
+    access via membership group. Make sure that the group allows only members
+    from within the organization, and that only invited members can join.
+
+You can add the following Cloud IAM snippet to the project policy:
+
+```
+    - auditLogConfigs:
+    - logType: DATA_READ
+    - logType: DATA_WRITE
+    service: cloudsql.googleapis.com
+
+    - members:
+      - group:xxxx-breakglass@groups.your-org.com
+    role: roles/cloudsql.admin
+
+    - members:
+      - group:xxxx-data-access@groups.your-org.com
+      - serviceAccount:xxxx-prod@xxxx.iam.gserviceaccont.com
+    role: roles/cloudsql.client
+```
+
+## Define MySQL users and passwords on your instance
+
+Because Cloud IAM acts as a primary athentication and authorization mechanism,
+we can consider MySQL usernames and passwords are a secondary access controls that
+can be used to further restrict access for reliability or safety
+purposes. For example, removing the ability of modifying tables from production
+users that don't need such a capability.
+
+The module, by default, creates users that:
+
+-   only allow connections from host ~cloudsqlproxy to ensure that nobody can
+    access data without connecting via the Cloud SQL Proxy.
+-   have randomly generated passwords, which can be stored in configuration
+    files. Such passwords can be considered as secure as API Keys rather than
+    strong credentials for access.
+
+To maintain the user list manageable, we suggest to consider MySQL users in a
+way similar to roles rather than individual human users. For example, a simple
+application could define the following roles:
+
+-   **admin**: root-like users with full permissions on all tables and data.
+    This can be used in breakglass emergencies, or as part of the release cycle
+    to perform structural modifications of the DB (e.g., add tables and columns)
+
+-   **app**: permissions required for running the application. Generally, it
+    will have read/write permissions on data, but no ability of performing
+    structural modifications to the DB.
+
+    *   Multiple roles for different apps can be defined if more protection
+        against accidental errors is desired.
+
+-   **readonly**: permissions required for reading data in the database for
+    debugging or customer support.
+
+*   Once users are created, you need to assign them
```

Comment:
(Later solution) Should we incorporate this SQL automatically?

Reply:
Good idea, I'll look into it for a followup pull request.

Reply:
(resolving for now, I can work on that improvement independently from this PR)